node/connection_base.js

```diff
@@ -266,6 +267,12 @@ TChannelConnectionBase.prototype.handleCallRequest = function handleCallRequest(
     }
 };
 
+TChannelConnectionBase.prototype.handlePingResponse =
+function handlePingResponse(res) {
+    var self = this;
+    self.pingResponseEvent.emit(self, res);
```

Comment:
Exposes version-specific frame shapes out of a connection-external api, connection exists to abstract over specifcs... the fact that you had to introduce a v2 require within `self_connection` should've been the clue.